prov/util/src/util_ep.c

```diff
@@ -247,8 +247,9 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 	if (util_domain->eq)
 		ofi_ep_bind_eq(ep, util_domain->eq);
 
-	/* TODO Figure out how to optimize this lock for rdm and msg endpoints */
-	ret = ofi_genlock_init(&ep->lock, OFI_LOCK_MUTEX);
+	ret = ofi_genlock_init(&ep->lock,
+			       ep->domain->threading != FI_THREAD_SAFE ?
+			       OFI_LOCK_NOOP : OFI_LOCK_MUTEX);
```

Comment:
Maybe misreading this (could have been an issue with the original commit) but the commit I think was supposed to target the ep_list_locks, this is the ep->lock itself which I think is a separate issue. I agree with changing the ep lock to a noop/mutex check but I don't think it belongs in this patch.

Reply:
I think it is clearer if you look at the files changed tab instead of commit by commit.